### DISCUSSION TITLE
slingserver 4.0 and slingerplayer not connect

### DISCUSSION THREAD

**Discussion by alfuxj** [GerryDazoo/Slinger#561] [Q&A] [answered]
i use slingserver 4.0 with config.ini , running  ubuntu  cloud server . the slingbox is remote in my home .
with vlc or other player work perfect .
i tried sligerplayer in windows and android with same result , (sling player error opening strem , incorrect url ? , server running ? )
the remote section work showing remote control . 
is something miss  in my configuration ? where I make mistake ?
i use http ://ip_cloud_server:8589
regards alfredo

**Comment by GerryDazoo** [comment]
Do you mean SlingERPlayer? Not slingplayer.
For sure the URL you typed is wrong. 
The drop down URL list of valid URLs should be populated automatically if SlingerPlayer and Slinger are on the same LAN. I suspect because you're using a cloud-based server you should be starting the Windows SlingerPlayer with this command line. I believe this was covered in the minimal documentation I wrote.
SlingerPlayer wan://ip_cloud_server:8589
That said, it's impossible to tell for sure what you're doing wrong without additional information. Please post your config.ini and ALL the server output from startup to the failed connection attempt. The Server should print out a list of valid URLs when it gets a bad request.

**Comment by alfuxj** [comment]
my configuration is : slingbox 500 is in italy . your server 4.0 running in cloud ubuntu server in germany , i use vlc or other player from china . all are on different lan .  all work perfect . i cannot connect use your android apk and same result with windows.  cannot start stream .  SlingerPlayer and Slinger are NOT on the same LAN.  i do not know if is possible use with this configuration .  Slinger ang slingbox500 are not on same LAN.
thank for your help.

**Reply by GerryDazoo** [reply]
You should be able to use the same URL that you use with VLC to connect. based on you config
http ://ip_cloud_server:8589/slingbox

**Reply by alfuxj** [reply]
i tried this http ://ip_cloud_server:8589/slingbox

but i get  a lot of Error. i try to make log .  
21:20:23 FFPlayer=Invalid ECC byte
21:20:23 FFPlayer= is not implemented. Update your FFmpeg version to the newest one from Git. If the problem still occurs, it means that your file has a feature which has not been implemented.

21:20:23 FFPlayer=If you want to help, upload a sample of this file to https://streams.videolan.org/upload/ and contact the ffmpeg-devel mailing list. (ffmpeg-devel@ffmpeg.org)

21:20:23 FFPlayer=unexpected packet_replic_size of %d

21:20:23 FFPlayer=invalid padsize %u at:%lld

21:20:24 FFPlayer=invalid padsize %u at:%lld

21:20:24 FFPlayer=Invalid ECC byte
21:20:24 FFPlayer= is not implemented. Update your FFmpeg version to the newest one from Git. If the problem still occurs, it means that your file has a feature which has not been implemented.

21:20:24 FFPlayer=If you want to help, upload a sample of this file to https://streams.videolan.org/upload/ and contact the ffmpeg-devel mailing list. (ffmpeg-devel@ffmpeg.org)

21:20:24 FFPlayer=packet_obj_size %d invalid

**Comment by alfuxj** [comment]
maybe i need open some port in cloud server?

**Reply by alfuxj** [reply]
SLINGBOX]
sbtype="350/500/M1/M2"
;sbtype="Solo/Pro/ProHD"
;sbtype="240"
password=DgK4xxxxxxxxxxxxx
; uncomment the next lines ipaddress and port with your local network info
; if you don't want the server to automatically find your
; slingbox on the local network. Needed if server and slingbox are
; not on the same LAN segment. Also removes requirement for the netifaces module
ipaddress=qddo.xxxxx.com
port=5201
; Repace with your finderid, If and only if you need remote access and don't
;have a static ip and you plan on using my service to access you server remotely.
; Please read the release notes. Most people don't need this.
;finderid=90ACxxxxxxxxxxxxxx

;If you've got more than one slingbox set name. THe server will use this when generating
; logs to make it easier to see what's going on.
;name=something

; Following are the default values
; Default resolution when server starts. You can change this and reconnect without having to restart the serve
r
; valid range = 1..15
Resolution=12

; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
FrameRate=30

;Min 50 Max 8000
VideoBandwidth=4000

; Min 0 (auto) max 63
VideoSmoothness=63

; Send Iframe every n seconds Max 30
;IframeRate=5

; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
AudioBitRate=32

; Video Source  0, 1, 2, 3 depending on your hardware corrosponds to one of
;Composite, Component, HDMI or Tuner.
;ProHD 0=Tuner, 1=composite 2=svideo 3=component
;Solo, M1 0=composite 1=Svideo 2=component
;500  0=composite, 1=component, 2=HDMI
; If you don't set this the code will use the last configired input
; I recommend setting this value. If not set correctly you'll often only see a black screen.
VideoSource = 2

[SERVER]
; local port number for the server to listen on for connections
port=8589
maxstreams=10
enableremote=yes
;maxremotestreams=2
;URLbase=Gerry






[REMOTE]
; see release notes for changing this if your remote doesn't work, But as a start make it
;the same as your configured VideoSource 0-3. See above.
code=1
; path to an external remote control definitions file, if the defaults
; aren't working for you or you want to change the format of the web page.
; use the supplied remote.txt as a starting point.
include=remotemio.txt
;include=skyhdRemote.html
root@ubuntude:/home/slingbox#

**Reply by alfuxj** [reply]
Version : 4.01 Running on Linux-5.15.0-67-generic-x86_64-with-glibc2.35 pid= 385
9 slingbox_server.py
Using config file config.ini
Connection Manager Running on port 8589 with 10 max streams using URL slingbox.
Building page for Slingbox
Reading Custom Remote definition from remotemio.txt
Starting Streamer Thread for Slingbox
starting up on  port 8589 8388608
Streamer Running:  10 config.ini SLINGBOX 8589 8589 8388608
8589 Checking for slingbox at ('xxx.duckdns.com', 5201) 0
8589 ('xxx.duckdns.com', 5201) OK
8589 Using slingbox at  ('xxx.duckdns.com', 5201)
Streamer:  8589 Waiting for first stream, flushing any IR requests that arrive w
hile not connected to slingbox
 * Serving Flask app 'slingbox_server'
 * Debug mode: off
Server IP address is xx.xxx.230.225
Broadcast Responder Running
03/22/2023, 12:33:52.860   Streaming request from ('1.1.69.214', 2980)  /
Error: Can't find streamer for 8589 in dict_keys(['/slingbox/8589'])
03/22/2023, 12:33:54.912   Streaming request from ('1.1.69.214', 3104)  /
Error: Can't find streamer for 8589 in dict_keys(['/slingbox/8589'])

**Comment by alfuxj** [comment, marked as answer]
http ://ip_cloud_server:8589/slingbox
yes this is the correct command . 
sometime  i can get connection and working. 
the problem is : from china the connection to europe is very hard chinese internet slowed-down cross-border internet traffic. and somethime lose packets.
this is the same problem i have with vlc . sometime start the most not start . 
infact for use with stable connection in my ubuntu server i made rtmp server . this connect your slingserver then transcode to rtmp . in my case from rtmp is more strong protocol or may elude better the greatwall firewall. i have stable flux and never lost connection sometime for more  days.
congratulation for your nice work .
thaks again

for me the tread is closed
